docs/pages/deploy-a-cluster/deployments/aws-starter-cluster-terraform.mdx

```diff
@@ -0,0 +1,714 @@
+limited permission scopes for AWS services. However, the initial cluster setup must be done by a user with a high
+level of AWS permissions.
+
+## Get the Terraform code
+
+Firstly, you'll need to clone the Teleport repo to get the Terraform code available on your system:
+
+```code
+$ git clone https://github.com/gravitational/teleport -b branch/v(=teleport.major_version=)
+# Cloning into 'teleport'...
+# remote: Enumerating objects: 106, done.
+# remote: Counting objects: 100% (106/106), done.
+# remote: Compressing objects: 100% (95/95), done.
+# remote: Total 61144 (delta 33), reused 35 (delta 11), pack-reused 61038
+# Receiving objects: 100% (61144/61144), 85.17 MiB | 4.66 MiB/s, done.
+# Resolving deltas: 100% (39141/39141), done.
+```
+
+Once this is done, you can change into the directory where the Terraform code is checked out and run `terraform init`:
+
+```code
+$ cd teleport/examples/aws/terraform/starter-cluster
+$ terraform init
+
+Initializing the backend...
+
+Initializing provider plugins...
+- Checking for available provider plugins...
+- Installing hashicorp/random v3.5.1...
+- Installed hashicorp/random v3.5.1 (signed by HashiCorp)
+- Installing hashicorp/aws v4.67.0...
+- Installed hashicorp/aws v4.67.0 (signed by HashiCorp)
+
+Terraform has been successfully initialized!
+
+You may now begin working with Terraform. Try running "terraform plan" to see
+any changes that are required for your infrastructure. All Terraform commands
+should now work.
+
+If you ever set or change modules or backend configuration for Terraform,
+rerun this command to reinitialize your working directory. If you forget, other
+commands will detect it and remind you to do so if necessary.
+```
+
+This will download the appropriate Terraform plugins needed to spin up Teleport using our
+reference code.
+
+## Set up variables
+
+Terraform modules use variables to pass in input. You can do this in a few ways:
+
+- on the command line to `terraform apply`
+- by editing the `vars.tf` file
+- by setting environment variables
+
+For this guide, we are going to make extensive use of environment variables. This is because it makes it easier for us
+to reference values from our configuration when running Teleport commands after the cluster has been created.
+
+Any set environment variable starting with `TF_VAR_` is automatically processed and stripped down by Terraform, so
+`TF_VAR_test_variable` becomes `test_variable`.
+
+We maintain an up-to-date list of the variables and what they do in the README.md file under [the
+`examples/aws/terraform/ha-autoscale-cluster` section of the Teleport repo](https://github.com/gravitational/teleport/blob/branch/v(=teleport.major_version=)/examples/aws/terraform/starter-cluster/README.md)
+but we'll run through an example list here.
+
+Things you will need to decide on:
+
+### region
+
+```code
+export TF_VAR_region="<Var name="us-west-2" />"
```

Comment:
All of the `code` snippets on this page with example commands should start with `$` so the docs engine renders them properly